graylog2-web-interface/src/components/grok-patterns/EditPatternModal.jsx

```diff
@@ -93,13 +97,18 @@ class EditPatternModal extends React.Component {
   };
 
   _saved = () => {
-    const { create } = this.props;
+    const { create, sendTelemetry } = this.props;
 
     this._closeModal();
 
     if (create) {
       this.setState({ name: '', pattern: '', sampleData: '', test_result: '' });
     }
+
+    sendTelemetry(TELEMETRY_EVENT_TYPE.GROK_PATTERN[create ? 'CREATED' : 'UPDATED'], {
```

Comment:
I think this should be tracked before anything else (setting state, closing modal) happens.